convert2rhel/actions/pre_ponr_changes/backup_system.py

```diff
@@ -63,3 +79,81 @@ def run(self):
 
         repo.backup_yum_repos()
         repo.backup_varsdir()
+
+
+class BackupPackageFiles(actions.Action):
+    id = "BACKUP_PACKAGE_FILES"
+
+    def run(self):
+        """Backup changed package files"""
+        loggerinst.task("Prepare: Backup package files")
+
+        super(BackupPackageFiles, self).run()
+
+        package_files_changes = self._parse_rpm_va()
+
+        for file in package_files_changes:
+            if file["status"] == "missing":
+                pass  # no action needed now
+            elif re.search(r"[5]", file["status"]):
+                # If the MD5 checksum differs, the content of the file differs
+                restorable_file = backup.NewRestorableFile(file["path"])
+                backup.backup_control.push(restorable_file)
+
+    def _parse_rpm_va(self):
+        """Get the output from rpm -Va command from generated during resolving system info
+        to get changes made to package files.
+        Return them as a list of dict, for example:
+        [{"status":"S5T", "file_type":"c", "path":"/etc/yum.repos.d/CentOS-Linux-AppStream.repo"}]
+        """
+        path = os.path.join(LOG_DIR, PRE_RPM_VA_LOG_FILENAME)
+        with open(path, "r") as f:
+            output = f.read()
+
+        output = output.strip()
+        lines = output.split("\n")
+        data = []
+
+        for line in lines:
+            parsed_line = self._parse_line(line.strip())
+            if parsed_line["path"]:
+                data.append(parsed_line)
+
+        return data
+
+    def _parse_line(self, line):
+        """Return {"status":"S5T", "file_type":"c", "path":"/etc/yum.repos.d/CentOS-Linux-AppStream.repo"}"""
+        ret = re.match(RPM_VA_REGEX, line)
+
+        if not ret:  # line not matching the regex
+            if line.strip() != "":
+                # Line is not empty string
+                loggerinst.debug("Skipping invalid output %s" % line)
+            return {"status": None, "file_type": None, "path": None}
```

Comment:
@SpyTec What about adding to the condition which checks if the path is not empty another one to check the path too. So it would like:

`if parsed_line["path"] and parsed_line["status"] :`
on this line: https://github.com/oamg/convert2rhel/pull/875/files#diff-5d146d6a82d3b246af03b014ec5d8aa5686de4e19801aaddf68d18d33fe5549eR119 (the one before appending the parsed_line)

Reply:
Sounds good to me!